### PR TITLE
[PREVIEW] PRO 3361 multiple applicants update documents page

### PR DESCRIPTION
**Pull request by tomxelliott**
Will rebase ontop of master once 3360 has been merged.

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: test/component/testDocumentsPage.js [hunk 1]
```diff
@@ -248,12 +257,112 @@ describe('documents-page', () => {
...
+                        'checklist3-item3',
+                        'coverLetter-codicils',
+                        'checkboxLabel-codicils',
+                        'checklist3-item4-Form205'
+                    ];
+                    testWrapper.testContent(done, excludeKeys, contentData);
+                });
+        });
+
+        it('test correct content loaded on the page when all executors name changed by deed poll', (done) => {
```
Comment (shepda) [inline test/component/testDocumentsPage.js]: I don't think we need this test as it's duplicated by the one above it

File: test/unit/testDocumentsPage.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+const initSteps = require('app/core/initSteps');
+const chai = require('chai');
+const expect = chai.expect;
+const steps = initSteps([`${__dirname}/../../app/steps/action/`, `${__dirname}/../../app/steps/ui`]);
+
+describe('ApplicantAliasReason', () => {
```
Comment (shepda) [inline test/unit/testDocumentsPage.js]: This test description should be `Documents`, not `ApplicantAliasReason`

File: test/unit/testDocumentsPage.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    const Documents = steps.Documents;
+
+    describe('getUrl()', () => {
+        it('should return the correct url', (done) => {
+            const url = Documents.constructor.getUrl();
+            expect(url).to.equal('/documents');
+            done();
+        });
+    });
+});
```
Comment (shepda) [inline test/unit/testDocumentsPage.js]: Can we also have a test for `handleGet()` for at least the addition of `ctx.executorsNameChangedByDeedPollList`

File: test/util/TestWrapper.js [hunk 4]
```diff
@@ -41,11 +41,11 @@ module.exports = class TestWrapper {
 
     testContent(done, excludeKeys = [], data) {
         const contentToCheck = cloneDeep(filter(this.content, (value, key) => !excludeKeys.includes(key) && key !== 'errors'));
-        this.substituteContent(data, contentToCheck);
+        const substitutedContent = this.substituteContent(data, contentToCheck);
```
Comment (shepda) [inline test/util/TestWrapper.js]: 👍

File: test/unit/testDocumentsPage.js (hunk 3 above)
Comment (tomxelliott) [reply test/unit/testDocumentsPage.js]: Unit testing this function raises an issue we have encountered in the past as a result of this call: const content = this.generateContent(ctx, formdata);
It causes the test in the payments file to fail. The executorsWrapper.executorsNameChangedByDeedPoll() has enough test coverage to make up for this.

**Pull request merged**